### DISCUSSION TITLE
Bridge error - [Error: download manifest failed (1) - Problem with the SSL CA cert (path? access rights?)]

### DISCUSSION THREAD

**Discussion by PeterJBassett** [iTwin/itwinjs-core#2862] [Q&A] [answered]
Hello, my application runs a bridge connection to iTwin using the  IModelBridge class. When the synchronisation runs I get the following error.
[Error: download manifest failed (1) - Problem with the SSL CA cert (path? access rights?)]

Please can you tell me if this is a known issue, and whether there is a solution?

From further investigation, these error messages exist in the following modules:
@bentley/imodeljs-native/imodeljs-darwin-x64/imodeljs.node
@bentley/imodeljs-native/imodeljs-darwin-x64/BeBlobDaemon

Can you provide any further information on why these error messages might be being raised?

**Comment by jchick-bentley** [comment]
Hi Peter,

Sorry to hear you are encountering this error.

There have been issues with firewalls that re-encrypt SSL traffic/self-signed certificates which work for the domain but then are untrusted by applications like Node.

If you think this scenario might apply to you, you can try the following:

// This must be done before any network calls to HTTPS endpoints.
Logger.logTrace(`${loggingCategory}.win_ca`, "Begin loading Windows certificate store...");
const loadedCerts: any[] = [];
ca({
inject: "+", // Let the Windows cert store augment built-in trust to defeat on-demand creation by Windows 10
ondata: loadedCerts,
});
like 1
const ca = require("win-ca");

If this doesn't help, please reply with a little more detail about what server/service you're trying to connect with and what operation is happening when you receive this error and any other details you think are relevant.  We'll be happy to try to solve this issue for you.

Regards,

Jason

**Comment by PeterJBassett** [comment, marked as answer]
Hello, Thank you for your reply.
I don't think this solution will help. My application does not run on Windows, it is running in a linux docker container. The application used to run ok in September, it is only recently that this error has started to occur.

The application is a node.js (v14) express server running in a node:14-bullseye-slim Docker container in our (FutureOn) kubernetes infrastructure.

Peter

**Reply by jchick-bentley** [reply]
Hi Peter,

Running a Node image (node:14.17.0-buster-slim and node:12.22.5-bullseye-slim), Connector developers have had success using the following to get a CA cert in their Docker file:

RUN apt-get update \
 && apt-get install -y --no-install-recommends ca-certificates
RUN update-ca-certificates

Can you try this?

Jason

**Comment by PeterJBassett** [comment]
Hello Jason

Thank you so much for your reply.
The solution you posted has successfully fixed the CA certificate problem.

Peter

**Reply by jchick-bentley** [reply]
Glad to hear it.  Thank you for confirming!
